Add Header and Side Navbar along with setting up Tailwind

dhairyaj opened the pull request:
Closes #14 

Implementation
1. Added Tailwind CSS
2. Added react-icons package
3. Modified folder structure and file names
4. Added Header and sidebar components
5. Added some basic styles and a placeholder for Logo and theme toggle functionality

Commit:
Add Header and Side Navbar along with setting up Tailwind

frankhart2018 commented on frontend/src/App.js:
```diff
@@ -1,13 +1,21 @@
+import LeftPanel from "./components/LeftPanel/LeftPanel";
+import RightPanel from "./components/RightPanel/RightPanel";
 
 function App() {
```
Let's use arrow functions to maintain consistency.

dhairyaj replied:
Done

frankhart2018 commented on frontend/src/components/LeftPanel/Header.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { BsFillSunFill, BsFillMoonFill } from "react-icons/bs";
+
+function Header() {
```
Arrow function

dhairyaj replied:
Done

frankhart2018 commented on frontend/src/components/LeftPanel/LeftPanel.js:
```diff
@@ -0,0 +1,21 @@
+import NavBar from './NavBar';
+import Header from './Header';
+
+function LeftPanel() {
```
Arrow function

dhairyaj replied:
Done

frankhart2018 commented on frontend/src/components/LeftPanel/NavBar.js:
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { FaUserAlt } from "react-icons/fa";
+function NavBar() {
```
Arrow function

dhairyaj replied:
Done

frankhart2018 commented on frontend/src/components/LeftPanel/NavBar.js:
```diff
@@ -0,0 +1,35 @@
+        {/* TODO:  Need to provide icons for following
+        Basic details, address, social media, work ex, education, awards, publications, certifications, skills, languages, interests, projects, references */}
+
+        <FaUserAlt />
```
Will there be links that move the view to that location as well? If so, then let's just put all the icons and links in an array of objects and map that here, hard-coded values make it inflexible to add more sections in the future.

The object can look like:

```js
{
    "href": "some-link",
    "icon": "icon component",
}
```

dhairyaj replied:
Seems like a good idea.

dhairyaj replied:
Done

frankhart2018 commented on frontend/src/components/RightPanel/RightPanel.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+function RightPanel() {
```
Arrow function

dhairyaj replied:
done

Commit:
Update to arrow functions

Commit:
Add Material UI and array for navbar items

dhairyaj commented:
Implementation changes in 2 updated commits
   - Formatted the code using prettier to maintain the code formatting standard
   - Modified the components using arrow functions
   - Added Material UI dependencies ( we will use material ui components and if needed tailwind)
   - Added a array of objects for the navigation menu items

frankhart2018 reviewed (approved):
LGTM
